xbmc/video/ContextMenus.cpp

```diff
@@ -231,10 +247,29 @@ bool CVideoPlay::Execute(const std::shared_ptr<CFileItem>& itemIn) const
   if (item.IsDVD() || item.IsCDDA())
     return MEDIA_DETECT::CAutorun::PlayDisc(item.GetPath(), true, true);
 #endif
-  SetPathAndPlay(item);
+  SetPathAndPlay(item, "");
   return true;
 };
 
+bool CVideoPlayUsing::IsVisible(const CFileItem& item) const
+{
+  const CPlayerCoreFactory& playerCoreFactory{CServiceBroker::GetPlayerCoreFactory()};
+  return (GetPlayers(playerCoreFactory, item).size() > 1) && VIDEO_UTILS::IsItemPlayable(item);
+}
+
+bool CVideoPlayUsing::Execute(const std::shared_ptr<CFileItem>& itemIn) const
+{
+  CFileItem item{itemIn->GetItemToPlay()};
+
+  const CPlayerCoreFactory& playerCoreFactory{CServiceBroker::GetPlayerCoreFactory()};
+  const std::vector<std::string> players{GetPlayers(playerCoreFactory, item)};
+  const std::string player{playerCoreFactory.SelectPlayerDialog(players)};
+  if (!player.empty())
+    SetPathAndPlay(item, player);
+
+  return true;
```

Comment:
wouldn't it make more sense to return true only if setpathandplay and false otherwise?

Reply:
Good point. Will change that.

Reply:
Also note that `CPlayListPlayer::Play` returns bool so theoretically you could improve the calls along the chain to also return the status code. Ofc this is far more work than the initial proposal.
You'd do 
`return !player.empty() && SetPathAndPlay(item, player)`

Reply:
Fair.

Reply:
First suggestion is in.